packages/contracts/contracts/registry/ens/ENSSubdomainRegistrar.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENS.sol";
+import "@ensdomains/ens-contracts/contracts/resolvers/Resolver.sol";
+import "../../core/component/Component.sol";
+
+/// @title A registrar for ENS subdomains
+/// @author Aragon Association - 2022
+/// @notice This contract registers ENS subdomains under a parent domain specified in the initialization process
+///         and maintains ownership of the subdomain since only the resolver address is set.
+///         This contract must either be the domain node owner or an approved operator of the node owner.
+///         During the subdomain initialization, the resolver specified in the parent domain is used.
+
+contract ENSSubdomainRegistrar is Component {
+    bytes4 internal constant REGISTRY_INTERFACE_ID =
```

Comment:
This should be tackled in https://aragonassociation.atlassian.net/browse/APP-524. 

`ENSSubdomainRegistrar ` is something that can clearly be reused. Rather than not using, `Component` we should rather revisit the design considerations behind `Component` and the use of `AdaptiveERC165` which was also suggested by Jorge (see https://aragonassociation.atlassian.net/browse/APP-558).
This is beyond the scope of this PR.

Reply:
Jorge's suggestions include using the minimal needed dependencies. Component inherits a lot of things, wondering why the ENS registrar needs all the properties of, say a plugin.  Do we need to register the registry?

Reply:
This was resolved regarding @Rekard0's point I assume, but I still wonder who may want to reuse this ENSSubdomainRegistrar so that we ?

I agree about reviewing the built-in features behind `Component` :+1:
[Created this task](https://aragonassociation.atlassian.net/browse/APP-619)

Reply:
One motivation is to save gas. However, we will deploy the registrar for our `dao.eth` domain only once or twice (per chain) so that this is not an issue.

Another motivation is to re-use `ENSSubdomainRegistrar` and allow developers to install it as a plugin/component to their Aragon DAO. This way, they can potentially manage and register their own subdomains for wallet or contract addresses. Thus, it is mandatory to make it a `Component`.

Generally, `Component` should be as convenient to use and as minimal as possible, but support all the features needed. ERC165 support is a mandatory feature (and I think we will use it heavily for our marketplace) and added to the `Component` design.
The fact that we are hesitant to use and trying to build around is concerning to me and suggests that we should revisit the component design (mainly `AdaptiveERC165` inside the `Component` contract as per our [JIRA issue APP-557](https://aragonassociation.atlassian.net/browse/APP-558) but also `MetaTxComponent`). 
If we don't want to use `Component` ourselves, why do we expect developers to use it?

Reply:
We will have this discussion as part of [APP-619](https://aragonassociation.atlassian.net/browse/APP-619), so this conversation can be resolved.